internal/testutil/acc/config_formatter.go

```diff
@@ -50,3 +57,184 @@ func sortStringMapKeys(m map[string]string) []string {
 	sort.Strings(keys)
 	return keys
 }
+func sortStringMapKeysAny(m map[string]any) []string {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ToSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func ClusterResourceHcl(projectID string, req *ClusterRequest) (configStr, clusterName string, err error) {
+	if req == nil {
+		req = new(ClusterRequest)
+	}
+	specs := req.ReplicationSpecs
+	if len(specs) == 0 {
+		specs = append(specs, ReplicationSpec(&ReplicationSpecRequest{}))
+	}
+	clusterName = req.ClusterNameExplicit
+	if clusterName == "" {
+		clusterName = RandomClusterName()
+	}
+	clusterTypeStr := "REPLICASET"
+	if req.Geosharded {
+		clusterTypeStr = "GEOSHARDED"
+	}
+
+	f := hclwrite.NewEmptyFile()
+	root := f.Body()
+	cluster := root.AppendNewBlock("resource", []string{"mongodbatlas_advanced_cluster", "cluster_info"}).Body()
+	clusterRootAttributes := map[string]any{
+		"project_id":     projectID,
+		"cluster_type":   clusterTypeStr,
+		"name":           clusterName,
+		"backup_enabled": req.CloudBackup,
+	}
+	if req.DiskSizeGb != 0 {
```

Comment:
knit: default values are spread though the function, e.g. for GEOSHARDED, etc.,   would be great to have them all together if possible.  not a big deal to leave as it is though